monitoring/alert/alert.go

```diff
@@ -0,0 +1,394 @@
+		Name: "projects/" + projectID,
+		// Filter:  "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+		// OrderBy: "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+	}
+	alertIt := alertClient.ListAlertPolicies(ctx, alertReq)
+	for {
+		resp, err := alertIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		b.Alerts = append(b.Alerts, &alertPolicy{resp})
+	}
+
+	// [START monitoring_alert_list_channels]
+	channelClient, err := monitoring.NewNotificationChannelClient(ctx)
+	if err != nil {
+		return err
+	}
+	channelReq := &monitoringpb.ListNotificationChannelsRequest{
+		Name: "projects/" + projectID,
+		// Filter:  "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+		// OrderBy: "", // See https://cloud.google.com/monitoring/api/v3/sorting-and-filtering.
+	}
+	channelIt := channelClient.ListNotificationChannels(ctx, channelReq)
+	for {
+		resp, err := channelIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		b.Channels = append(b.Channels, &channel{resp})
+	}
+	// [END monitoring_alert_list_channels]
+	e := json.NewEncoder(w)
+	if err := e.Encode(b); err != nil {
+		return err
+	}
+	return nil
+}
+
+// alertPolicy is a wrapper around the AlertPolicy proto to
+// ensure JSON marshaling/unmarshaling works correctly.
+type alertPolicy struct {
+	*monitoringpb.AlertPolicy
+}
+
+// channel is a wrapper around the NotificationChannel proto to
+// ensure JSON marshaling/unmarshaling works correctly.
+type channel struct {
+	*monitoringpb.NotificationChannel
+}
+
+// backup is used to backup and restore a project's policies.
+type backup struct {
+	ProjectID string
+	Alerts    []*alertPolicy
+	Channels  []*channel
+}
+
+func (a *alertPolicy) MarshalJSON() ([]byte, error) {
+	m := &jsonpb.Marshaler{EmitDefaults: true}
+	b := new(bytes.Buffer)
+	m.Marshal(b, a.AlertPolicy)
+	return b.Bytes(), nil
+}
+
+func (a *alertPolicy) UnmarshalJSON(b []byte) error {
+	u := &jsonpb.Unmarshaler{}
+	a.AlertPolicy = new(monitoringpb.AlertPolicy)
+	return u.Unmarshal(bytes.NewReader(b), a.AlertPolicy)
+}
+
+func (c *channel) MarshalJSON() ([]byte, error) {
+	m := &jsonpb.Marshaler{}
+	b := new(bytes.Buffer)
+	m.Marshal(b, c.NotificationChannel)
+	return b.Bytes(), nil
+}
+
+func (c *channel) UnmarshalJSON(b []byte) error {
+	u := &jsonpb.Unmarshaler{}
+	c.NotificationChannel = new(monitoringpb.NotificationChannel)
+	return u.Unmarshal(bytes.NewReader(b), c.NotificationChannel)
+}
+
+// [END monitoring_alert_backup_policies]
+
+// [START monitoring_alert_restore_policies]
+// [START monitoring_alert_create_policy]
+// [START monitoring_alert_create_channel]
+// [START monitoring_alert_update_channel]
+
+// restorePolicies updates the project with the alert policies and
+// notification channels in r.
+func restorePolicies(w io.Writer, projectID string, r io.Reader) error {
```

Comment:
no region tags here?

Reply:
Keeping the region tags outside the function.